MDANSE_GUI/Src/MDANSE_GUI/InputWidgets/TrajectoryFilterWidget.py

```diff
@@ -0,0 +1,1463 @@
+
+    Attributes
+    ----------
+    _bounds_off : Set[string]
+        The settings corresponding to the frequency bounds 'off' state.
+        When these settings are used, the filter frequency is unbounded.
+    _bounds_on : Set[string]
+        The settings corresponding to the frequency bounds 'on' state.
+        When these settings are used, the filter frequency is bounded.
+    """
+
+    # Bounds behaviour corresponding to attenuation settings
+    _bounds_off = {"lowpass", "highpass"}
+    _bounds_on = {"bandpass", "bandstop"}
+
+    # Signal: emitted when frequency bounds are enabled
+    _frequency_bounded = Signal(bool)
+
+    def __init__(
+        self,
+        parent_attributes: dict,
+        schema: dict,
+        render_func: Callable,
+        flags: set = set(),
+    ):
+        """
+        Parameters
+        ----------
+        parent_attributes : dict
+            Dictionary of attributes belonging to the parent filter designer widget.
+        schema : dict
+            Dictionary representing the schema for the filter-specific settings.
+        render_func : Callable
+            Filter designer function to call when attributes have been updated.
+            Function re-renders the filter designer graph with updated attributes.
+        flags : set
+            The set of flags associated with the current filter that can be used to invoke certain rules about settings.
+        """
+        super().__init__(parent_attributes, schema, render_func, flags)
+
+        # Generate an extra index for the frequency bound widget in the grid layout
+        last_index = self.indices[-1]
+        self.indices.append(((last_index[0][0] + 1, 0), (last_index[1][0] + 1, 1)))
+
+        # Connection: when the attenuation type requires/doesn't require frequency bounds, toggle the bounds widget on/off
+        self._frequency_bounded.connect(self.toggle_bound_frequencies)
+
+    def get_frequency_bounds(self) -> list:
+        """Create a list representing the upper and lower bounds of the filter critical frequencies
+
+        Returns
+        -------
+        list :
+            List of length 2 containing the critical frequency bounds.
+        """
+        cutoff = self.retrieve_widget("cutoff_freq")
+        bound = self.retrieve_widget("bound_freq")
+
+        return np.array(
+            sorted(
+                [
+                    cutoff.value(),
+                    bound.value(),
+                ]
+            )
+        ).tolist()
+
+    def toggle_bound_frequencies(self, on: bool = True) -> None:
+        """Toggle the pair of critical frequency inputs on/off.
+
+        Parameters
+        --------
+        on : bool
+            If true, both inputs for upper and lower frequency bounds are enabled, else only one input is enabled
+        """
+        bounds = self.retrieve_widget("bound_freq")
+        if bounds and on:
+            bounds.setEnabled(True)
+            return
+        bounds.setEnabled(False)
```

Comment:
```suggestion
        if bounds:
            bounds.setEnabled(on)
```
Otherwise if `not bounds`, what are you `setEnabled` on?